utilities/cli/util.py

```diff
@@ -187,20 +192,12 @@ def check_nvidia_ctk(min_version: str = "1.12.0", recommended_version: str = "1.
         match = re.search(r"(\d+\.\d+\.\d+)", output)
         if match:
             version = match.group(1)
-
             try:
-                from packaging import version as ver
-
-                version_check = ver.parse(version) < ver.parse(min_version)
-            except ImportError:
-
-                def parse_version(v):
-                    try:
-                        return tuple(map(int, v.split(".")))
-                    except ValueError:
-                        return (10, 0, 0)
```

Comment:
consider keeping this try... except ValueError, for use cases such as `list(map(int, '1.2.dev0'.split('.')))` it'll be a value error

Reply:
Good suggestion, updated as follows for the general case:
- Added stricter regex check to get MAJOR.MINOR.PATCH... from [semantic version](https://semver.org/) and strip trailing postfix without error, i.e. `1.2.3+dev0` is a valid case
- Added test coverage to validate behavior

In my opinion we can consider `1.2.dev0` as a valid ValueError because it does not have a valid PATCH version as defined by semantic versioning. We know that we can get a semantic version for the two components where this check is currently applied (`python3-dev` and `nvidia-container-toolkit`), so this seems like a safe assumption for now.